lib/drowsy/relation.rb

```diff
@@ -11,9 +11,20 @@ def initialize(klass)
     @klass = klass
     @connection = klass.connection
     @uri_template = klass.uri
+    @http_method = :get
     @params = {}
   end
 
+  # defines get, put, post, patch, delete
+  Drowsy::Http::METHODS.each do |http_method|
+    define_method http_method do |uri_template = nil, **params|
+      where(params)
+        .with_http_method(http_method)
+        .with_uri(uri_template)
+        .to_a
+    end
+  end
```

Comment:
this will result in some weird behavior for put, post, patch, delete... also, i probably need to change the innards of how `fetch` works because right now it's wrapping all results in an Array and then allowing the specific Relation methods to decide how many records it wants to return. if i don't change that then these custom requests can only be used to get back collections of results... the change would be checking if the response is an array or a hash and returning accordingly, it's just annoying because it pushes some weird instance_or_collection behavior into the guts of relation that i was trying to avoid...

decisions...